### PR TITLE
Add Python 3.5 support

### PR DESCRIPTION
**Pull request by onovy**
Fixed unit tests fail on Python 3.5
Added Python 3.5 in Travis CI
Mark Python 3.5 as supported in setup.py

### CODE REVIEW

File: tests/test_future/test_bytes.py [hunk 1]
```diff
@@ -534,13 +536,14 @@ def test_maketrans(self):
...
 
-    @unittest.expectedFailure
     def test_mod(self):
         """
         From Py3.5 test suite (post-PEP 461).
 
         The bytes mod code is in _PyBytes_Format() in bytesobject.c in Py3.5.
         """
+        if sys.version_info[:2] < (3, 5):
+            raise unittest.SkipTest('bytes % not yet implemented on Py < 3.5')
```
Comment (Daenyth) [inline tests/test_future/test_bytes.py]: You could also do `@unittest.skipIf(bool, msg)`

**Comment by jdufresne** [comment]
You could add Python 3.6 as well as it has since been released.

**Comment by Uzume** [comment]
Pull request #300 also tries to add 3.5 in addition to 3.6, etc.

**Comment by jmadler** [comment]
Can you rebase please to bring the latest TravisCI infra fixes in, and get this working?

**Comment by onovy** [comment]
all changes already in master. thanks.

**Pull request closed without merging**